tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/JoinClauseTest.java

```diff
@@ -150,7 +155,16 @@ public void testNegativeScenarios() {
         validateError(negativeResult, i++, "missing on keyword", 309, 1);
         validateError(negativeResult, i++, "undefined symbol 'dept'", 329, 24);
         validateError(negativeResult, i++, "missing equals keyword", 330, 1);
-        validateError(negativeResult, i, "missing identifier", 330, 1);
+        validateError(negativeResult, i++, "missing identifier", 330, 1);
+        validateError(negativeResult, i++, "outer join must be declared with var", 353, 19);
+        validateError(negativeResult, i++, "undefined symbol 'dept'", 357, 19);
+        validateError(negativeResult, i++, "invalid operation: type 'Person?' does not support field access", 374, 16);
+        validateError(negativeResult, i++, "incompatible types: expected 'int', found 'other'", 389, 59);
+        validateError(negativeResult, i++, "invalid operation: type 'Person?' does not support field access", 389, 59);
+        validateError(negativeResult, i++, "invalid operation: type 'Person?' does not support field access", 395, 22);
+        validateError(negativeResult, i++, "order by not supported for complex type fields, order key should belong" +
```

Comment:
Wondering why we get this error? Seems at that point, `person` type is `other`. Can you please check

Reply:
This is because `person` is Person? which is expected. Hence, resolving